src/main/java/org/jabref/logic/formatter/Formatters.java

```diff
@@ -102,6 +103,9 @@ private Formatters() {
         if (modifier.startsWith(RegexFormatter.KEY)) {
             String regex = modifier.substring(RegexFormatter.KEY.length());
             return Optional.of(new RegexFormatter(regex));
+        } else if (modifier.matches(TruncateFormatter.KEY + "\\d+")) {
```

Comment:
Special check because it requires numeric input.

Reply:
I think it would also make sense to extract this to a Pattern and call compile on it
https://www.geeksforgeeks.org/pattern-compilestring-method-in-java-with-examples/